build/chip/chip_test_suite.gni

```diff
@@ -99,7 +99,104 @@ template("chip_test_suite") {
       public_deps += [ "${chip_root}/src/platform/logging:force_stdio" ]
     }
   }
+  if (chip_link_tests) {
+    tests = []
+
+    if (defined(invoker.test_sources)) {
+      foreach(_test, invoker.test_sources) {
+        _test_name = string_replace(_test, ".cpp", "")
+
+        pw_test(_test_name) {
```

Comment:
I'm trusting that this part is correct; I don't know enough about pw_test to tell.

Reply:
At the very least, it seems to be working in practice - I've verified that when a suite uses pw_unit_test, running `./scripts/tests/all_tests.sh` will indeed cause the unit tests to get executed. The `pw_test` gn template (defined in `third_party/pigweed/repo/pw_unit_test/test.gni`) appears to be doing more or less the same thing as our very own `chip_test`, so it should be correct, unless I'm missing something.